core-aam/aria-hidden_value_changes-manual.html

```diff
@@ -0,0 +1,77 @@
+<!doctype html>
+<html>
+  <head>
+    <title>aria-describedby value changes</title>
+    <meta content="text/html; charset=utf-8" http-equiv="Content-Type"/>
+    <link rel="stylesheet" href="/wai-aria/scripts/manual.css">
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="/wai-aria/scripts/ATTAcomm.js"></script>
+    <script>
+    setup({explicit_timeout: true, explicit_done: true });
+
+    var theTest = new ATTAcomm(
+    {
+   "steps" : [
+      {
+         "attribute" : "aria-hidden",
+         "element" : "test",
+         "title" : "step 1",
+         "type" : "attribute",
+         "value" : "true"
+      },
+      {
+         "element" : "test",
+         "test" : {
+            "ATK" : [
+               [
+                  "event",
+                  "type",
+                  "is",
+                  "object:property-change:aria-hidden"
```

Comment:
I'm afraid this is not a thing. You cannot just create new properties. The available properties (and their names) for which you can do `object:property-change:foo` can be found in the ATK docs: https://docs.gtk.org/atk/class.Object.html#properties. As you'll see, `aria-hidden` is not there.

I've not yet found an `object:attributes-changed` signal in the ATK docs, but looking in Accerciser, it is listed in the available AT-SPI events one can monitor. Part of me suspects this is a case of ATK and AT-SPI accidentally falling out of alignment, and I need to verify that before you change your mapping to a signal that doesn't actually exist. :smile:  And if it does exist in ATK, my guess is you will have 0 implementations in browsers. Anyway, let me get back to you on this particular thing.

In the meantime.... What's what correct signal?  That depends on whether or not the user agent prunes hidden objects from the accessibility tree or not. I think most do these days.

Last time I checked, user agents don't fire property-change/attribute-change/et al. events on objects being destroyed (pruned). Thus, going from `aria-hidden=false` (or undefined/absent) to `aria-hidden=true` should result in things like children-removed, element-destroyed/state-changed:defunct, etc.

Similarly, if if hidden objects are pruned,  an element which had `aria-hidden=true` and then had that property removed/set to `false` would go from not being in the accessibility tree to being inserted into the accessibility tree. Thus I'd expect children-added, element-created, etc. Not property-change/attribute-change/et al.  After all, you cannot -- or rather, should not; I think Chromium sometimes does -- "change" attributes or properties on an accessible object that didn't exist prior to the property changing on the element.

This is a long way of saying that -- pending confirmation in user agents -- the AXAPI mapping looks right, and the other three do not. If I'm right, and the object is being pruned/re-added, I think the events you'll want are in the second table under https://w3c.github.io/core-aam/#mapping_events_visibility . What you have for AXAPI already matches that btw.

If instead the object is left in the tree, for ATK you'll want `object:state-changed:visible`.

Reply:
Update: `object:attributes-changed` is not a thing. It should hopefully become one in a few weeks. https://gitlab.gnome.org/GNOME/at-spi2-core/-/issues/153

Reply:
thanks joanie, for the detailed explanation and the follow up! I made this PR two years ago I must have been copy pasta-ing from the other event tests without realizing what I was doing. Thank you for the thorough check!!

Reply:
Ok I'm moving aria-hidden to a different PR in order to test on windows before submitting!